bot/exts/evergreen/emoji_count.py

```diff
@@ -17,73 +19,71 @@ class EmojiCount(commands.Cog):
+        return embed, msg
 
     @staticmethod
-    def generate_invalid_embed(ctx: commands.Context) -> discord.Embed:
-        """Genrates error embed."""
+    def generate_invalid_embed(ctx: commands.Context) -> Tuple[discord.Embed, str]:
+        """Generates error embed."""
         embed = discord.Embed(
             color=Colours.soft_red,
             title=random.choice(ERROR_REPLIES)
         )
+        msg = []
 
-        emoji_dict = {}
+        emoji_dict = defaultdict(list)
         for emoji in ctx.guild.emojis:
-            emoji_dict[emoji.name.split("_")[0]] = []
+            emoji_dict[emoji.name.split("_")[0]].append(emoji)
 
-        error_comp = ', '.join(key for key in emoji_dict.keys())
-        embed.description = f"These are the valid categories\n```{error_comp}```"
-        return embed
+        error_comp = ', '.join(emoji_dict)
+        msg.append(f"These are the valid categories\n```{error_comp}```")
+        return embed, msg
 
-    def emoji_list(self, ctx: commands.Context, categories: dict) -> Dict:
-        """Generates an embed with the emoji names and count."""
-        out = {category: [] for category in categories}
+    @commands.command(name="emojis_count", aliases=["ec"])
+    async def emojis_count(self, ctx: commands.Context, *, category_query: str = None) -> None:
+        """Returns embed with emoji category and info given by the user."""
+        emoji_dict = defaultdict(list)
 
+        if not ctx.guild.emojis:
+            await ctx.send("No emojis found.")
+            return
+        log.trace(f"Emoji Category {'' if category_query else 'not '}provided by the user")
         for emoji in ctx.guild.emojis:
-            category = emoji.name.split('_')[0]
-            if category in out:
-                out[category].append(emoji)
-        return out
-
-    @commands.command(name="emoji_count", aliases=["ec"])
-    async def ec(self, ctx: commands.Context, *, emoji: str = None) -> Optional[str]:
-        """Returns embed with emoji category and info given by the user."""
-        emoji_dict = {}
+            emoji_category = emoji.name.split("_")[0]
 
-        for a in ctx.guild.emojis:
-            if emoji is None:
-                log.trace("Emoji Category not provided by the user")
-                emoji_dict.update({a.name.split("_")[0]: []})
-            elif a.name.split("_")[0] in emoji:
-                log.trace("Emoji Category provided by the user")
-                emoji_dict.update({a.name.split("_")[0]: []})
+            if category_query is not None and emoji_category not in category_query:
+                continue
 
-        emoji_dict = self.emoji_list(ctx, emoji_dict)
+            emoji_dict[emoji_category].append(emoji)
 
         if len(emoji_dict) == 0:
-            embed = self.generate_invalid_embed(ctx)
+            log.trace("Invalid name provided by the user")
+            embed, msg = self.generate_invalid_embed(ctx)
```

Comment:
Instead of sending the context, you could only send the emojis (ctx.guild.emojis) as that is what is actually required.